### PR TITLE
Cleanup jsonpickle

### PR DESCRIPTION
**Pull request by marmistrz**
This whole PR would have taken 10x less time if we were using a language which has normal type checking and not duck-typing everywhere :/

- [x] `golem/core/compress.py`
  * The load and save functions seem to be used only in the tests
- [x] `golem/task/taskpreset.py`
  * dead import
- [x] `golem/core/simpleserializer.py`
  * This one is used in `golem/core/prochelper.py` to save snapshots containing the information about all application instance processes. The values stored are: `float` (`timestamp`), `int` and `NoneType`. This can be safely serialized using the `json` module instead.
- [x] `golem/model.py `
  * Used for python_value and db_value of JsonField. The field is used in the following classes:
    * `Payment`, field `dialog`, which is a dictionary containing the following keys
       * `'fee'` - of type `int`
       * `'block_hash'` - of type `str`
       * `'block_number'` - of type `int`
       * `'check'` - of type `bool`
    * `ExpectedIncome`, field `sender_node_details` of class `golem.network.p2p.node.Node` - this needs proper serialization
    * `TaskPreset`, field `data`
- [x] `golem/diag/service.py`
Here only dumps is used, so no risk. But why don't we simply use json?
- [x] `golem/interface/client/tasks.py`
Here a loads in while reading a task from file.

### CODE REVIEW

**Comment by marmistrz** [comment]
All the automatic tests pass on Circle and should pass on the other CIs. 

What's left is fixing the linter messages (hence the fail status on Circle), including 71 (sic!) "line-too-longs", mostly from legacy code.

File: tests/golem/docker/test_docker_luxrender_task.py [hunk 1]
```diff
@@ -76,7 +76,7 @@ def set_root_dir(p, new_root_dir=golem_dir):
 
         return task_def
 
-    def _test_task(self):# -> LuxTask():
+    def _test_task(self) -> LuxTask:
```
Comment (mfranciszkiewicz) [inline tests/golem/docker/test_docker_luxrender_task.py:79]: This function returns a tuple

File: circle.yml [hunk 2]
```diff
@@ -43,7 +43,7 @@ test:
     - pip3 install coverage codecov
   override:
     - ./lintdiff.sh pylint --disable=R apps golem gui scripts setup_util '*.py'
-    - ./lintdiff.sh pylint --disable=R,protected-access tests
+    - ./lintdiff.sh pylint --disable=R,protected-access,too-many-lines tests
```
Comment (mfranciszkiewicz) [inline circle.yml]: This change belongs to a separate PR.

File: tests/golem/test_client.py [hunk 3]
```diff
@@ -638,12 +732,18 @@ def test_enqueue_new_task_dict(self, *_):
                 '/Users/user/Desktop/folder/model.mesh',
                 '/Users/user/Desktop/folder/stylized_levi.blend'
             ],
-            'name': 'Golem Task 17:41:45 GMT+0200 (CEST)',
-            'type': 'blender',
-            'timeout': '09:25:00',
-            'subtasks': '6',
-            'subtask_timeout': '4:10:00',
-            'bid': '0.000032',
+            'name':
```
Comment (mfranciszkiewicz) [inline tests/golem/test_client.py]: After the change, this part is (IMO) much less readable. Please do not put values in a new line.

**Review by mfranciszkiewicz — COMMENTED** [review]
Please avoid making changes to code parts unrelated to the issue you are working on (i.e. reformatting, annotating).

File: tests/golem/docker/test_docker_luxrender_task.py (hunk 1 above)
Comment (marmistrz) [reply tests/golem/docker/test_docker_luxrender_task.py:79]: I added `@enforce.runtime_validation` and all the tests pass.

File: tests/golem/test_client.py (hunk 3 above)
Comment (marmistrz) [reply tests/golem/test_client.py]: I missed that, thanks!
